File: test-system.ts
import { StepFunctionsClient, StartExecutionCommand, DescribeExecutionCommand } from '@aws-sdk/client-sfn';
import { S3Client, PutObjectCommand, GetObjectCommand } from '@aws-sdk/client-s3';

const stepFunctionsClient = new StepFunctionsClient({ region: 'us-east-1' });
const s3Client = new S3Client({ region: 'us-east-1' });

interface TestCase {
  name: string;
  transcript: string;
  expectedSeverity: string;
}

const testCases: TestCase[] = [
  {
    name: 'High Severity - Critical Outage',
    transcript: 'This is a critical emergency! Our entire production system has been down for 2 hours. We are losing thousands of dollars per minute and need immediate assistance!',
    expectedSeverity: 'HIGH'
  },
  {
    name: 'Medium Severity - Frustrated Customer',
    transcript: 'I am very frustrated with your service. My internet has been slow for days and I have called multiple times. This is disappointing and needs to be resolved.',
    expectedSeverity: 'MEDIUM'
  },
  {
    name: 'Low Severity - General Inquiry',
    transcript: 'Hello, I would like to inquire about upgrading my current plan. Could you please provide information about your premium packages?',
    expectedSeverity: 'LOW'
  }
];

async function testStepFunctions(stateMachineArn: string): Promise<void> {
  console.log('🧪 Testing Step Functions with real Transcribe + Bedrock integration...\n');

  for (const testCase of testCases) {
    console.log(`Testing: ${testCase.name}`);
    
    const input = {
      jobName: \`test-\${Date.now()}-\${Math.random().toString(36).substr(2, 9)}\`,
      s3Uri: 's3://voice-to-sap-input-183631346754/test-audio.wav',
      transcript: testCase.transcript
    };

    try {
      // Start execution
      const startCommand = new StartExecutionCommand({
        stateMachineArn,
        name: \`test-execution-\${Date.now()}\`,
        input: JSON.stringify(input)
      });

      const startResponse = await stepFunctionsClient.send(startCommand);
      console.log(\`✅ Started execution: \${startResponse.executionArn}\`);

      // Wait and check status
      await new Promise(resolve => setTimeout(resolve, 5000));

      const describeCommand = new DescribeExecutionCommand({
        executionArn: startResponse.executionArn!
      });

      const describeResponse = await stepFunctionsClient.send(describeCommand);
      console.log(\`Status: \${describeResponse.status}\`);

      if (describeResponse.status === 'SUCCEEDED') {
        console.log(\`✅ Expected: \${testCase.expectedSeverity}\`);
        console.log(\`📄 Output: \${describeResponse.output}\`);
      } else if (describeResponse.status === 'FAILED') {
        console.log(\`❌ Execution failed: \${describeResponse.error}\`);
      } else {
        console.log(\`⏳ Still running: \${describeResponse.status}\`);
      }

    } catch (error) {
      console.error(\`❌ Error testing \${testCase.name}:\`, error);
    }

    console.log('---\n');
  }
}

async function uploadTestFile(): Promise<void> {
  console.log('📁 Uploading test voice file to trigger pipeline...\n');

  const testData = {
    metadata: {
      uploadedAt: new Date().toISOString(),
      testFile: true
    },
    transcript: 'This is urgent! Our system is completely broken and not working. We need immediate help!'
  };

  try {
    const command = new PutObjectCommand({
      Bucket: 'voice-to-sap-input-183631346754',
      Key: \`test-voice-\${Date.now()}.json\`,
      Body: JSON.stringify(testData),
      ContentType: 'application/json'
    });

    await s3Client.send(command);
    console.log('✅ Test file uploaded successfully!');
    console.log('🔄 Pipeline should start automatically via EventBridge...');

  } catch (error) {
    console.error('❌ Error uploading test file:', error);
  }
}

async function main(): Promise<void> {
  const stateMachineArn = 'arn:aws:states:us-east-1:183631346754:stateMachine:VoiceToSapStateMachine';

  console.log('🚀 Voice-to-SAP System Test Suite');
  console.log('==================================\n');

  const choice = process.argv[2];

  switch (choice) {
    case 'stepfunctions':
      await testStepFunctions(stateMachineArn);
      break;
    case 'upload':
      await uploadTestFile();
      break;
    default:
      console.log('Usage:');
      console.log('  npm run test:stepfunctions  - Test Step Functions directly');
      console.log('  npm run test:upload         - Upload file to trigger pipeline');
      break;
  }
}

main().catch(console.error);

File: test_system.py
#!/usr/bin/env python3
"""
Test script for the Voice-to-SAP ingestion system
"""

import boto3
import json
import time
from datetime import datetime

def test_step_functions():
    """Test the Step Functions workflow with mock data"""
    
    # Initialize AWS clients
    sfn_client = boto3.client('stepfunctions')
    
    # Mock input data
    test_input = {
        "jobName": f"test-job-{int(time.time())}",
        "s3Uri": "s3://voice-to-sap-input/test-voice-file.wav",
        "transcript": "Hello, I'm calling because my internet service has been down for 3 hours. This is urgent as I work from home and need connectivity immediately. I'm very frustrated with this outage."
    }
    
    try:
        # Get the state machine ARN (you'll need to update this after deployment)
        state_machine_arn = "arn:aws:states:us-east-1:123456789012:stateMachine:VoiceToSapStateMachine"
        
        print("Starting Step Functions execution...")
        print(f"Input: {json.dumps(test_input, indent=2)}")
        
        # Start execution
        response = sfn_client.start_execution(
            stateMachineArn=state_machine_arn,
            name=f"test-execution-{int(time.time())}",
            input=json.dumps(test_input)
        )
        
        execution_arn = response['executionArn']
        print(f"Execution started: {execution_arn}")
        
        # Monitor execution
        while True:
            status_response = sfn_client.describe_execution(executionArn=execution_arn)
            status = status_response['status']
            
            print(f"Execution status: {status}")
            
            if status in ['SUCCEEDED', 'FAILED', 'TIMED_OUT', 'ABORTED']:
                break
                
            time.sleep(10)
        
        if status == 'SUCCEEDED':
            print("✅ Execution completed successfully!")
            print(f"Output: {status_response.get('output', 'No output')}")
        else:
            print(f"❌ Execution failed with status: {status}")
            print(f"Error: {status_response.get('error', 'Unknown error')}")
            
    except Exception as e:
        print(f"Error testing Step Functions: {str(e)}")

def test_lambda_function():
    """Test the severity assignment Lambda function locally"""
    
    # Import the Lambda function
    import sys
    sys.path.append('./lambda/severity-assignment')
    from index import lambda_handler
    
    # Test cases
    test_cases = [
        {
            "name": "High severity - urgent negative",
            "event": {
                "Body": {
                    "content": [{"text": "Sentiment: negative, Intensity: 9"}]
                },
                "transcript": "This is urgent! My system is completely broken and not working at all!"
            }
        },
        {
            "name": "Medium severity - complaint",
            "event": {
                "Body": {
                    "content": [{"text": "Sentiment: negative, Intensity: 6"}]
                },
                "transcript": "I'm frustrated with this service. It's terrible and disappointing."
            }
        },
        {
            "name": "Low severity - positive",
            "event": {
                "Body": {
                    "content": [{"text": "Sentiment: positive, Intensity: 3"}]
                },
                "transcript": "Thank you for your help. Everything is working fine now."
            }
        }
    ]
    
    print("Testing Lambda function locally...")
    print("=" * 50)
    
    for test_case in test_cases:
        print(f"\nTest: {test_case['name']}")
        result = lambda_handler(test_case['event'], {})
        print(f"Severity: {result.get('severity')}")
        print(f"Sentiment: {result.get('sentiment')}")
        print(f"Intensity: {result.get('intensity')}")

if __name__ == "__main__":
    print("Voice-to-SAP System Test")
    print("=" * 30)
    
    choice = input("Choose test:\n1. Lambda function (local)\n2. Step Functions (AWS)\nEnter choice (1 or 2): ")
    
    if choice == "1":
        test_lambda_function()
    elif choice == "2":
        test_step_functions()
    else:
        print("Invalid choice")
